src/ResourceManager/EventGrid/Commands.EventGrid/Topic/UpdateAzureEventGridTopic.cs

```diff
@@ -0,0 +1,134 @@
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.EventGrid.Models;
+
+namespace Microsoft.Azure.Commands.EventGrid
+{
+    [Cmdlet(VerbsData.Update, EventGridTopicVerb, SupportsShouldProcess = true, DefaultParameterSetName = TopicNameParameterSet), OutputType(typeof(PSTopic))]
+    public class UpdateAzureEventGridTopic : AzureEventGridCmdletBase
+    {
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 0,
+            HelpMessage = "Resource Group Name.",
+            ParameterSetName = TopicNameParameterSet)]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        [Alias(AliasResourceGroup)]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 1,
+            HelpMessage = "EventGrid Topic Name.",
+            ParameterSetName = TopicNameParameterSet)]
+        [ValidateNotNullOrEmpty]
+        [Alias("TopicName")]
+        public string Name { get; set; }
+
+        [Parameter(Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 0,
+            HelpMessage = "EventGrid Topic ResourceID.",
+            ParameterSetName = ResourceIdEventSubscriptionParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(Mandatory = true,
+            ValueFromPipeline = true,
+            Position = 0,
+            HelpMessage = "EventGrid Topic object.",
+            ParameterSetName = TopicInputObjectParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public PSTopic InputObject { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            Position = 2,
+            HelpMessage = "Hashtable which represents resource Tags.",
+            ParameterSetName = TopicNameParameterSet)]
+        [Parameter(
+            Mandatory = true,
+            Position = 1,
+            HelpMessage = "Hashtable which represents resource Tags.",
+            ParameterSetName = ResourceIdEventSubscriptionParameterSet)]
+        [Parameter(
+            Mandatory = true,
+            Position = 1,
+            HelpMessage = "Hashtable which represents resource Tags.",
+            ParameterSetName = TopicInputObjectParameterSet)]
+        public Hashtable Tag { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            Dictionary<string, string> newTagDictionary = TagsConversionHelper.CreateTagDictionary(this.Tag, true);
+
+            if (this.ShouldProcess(this.Name, $"Update topic {this.Name} in Resource Group {this.ResourceGroupName}"))
+            {
+                string resourceGroupName = string.Empty;
+                string topicName = string.Empty;
+
+                if (!string.IsNullOrEmpty(this.ResourceId))
```

Comment:
@kalyanaj this chain of if-else-statements should go before the `ShouldProcess`, and then the `ShouldProcess` message should use `topicName` and `resourceGroupName` since those variables would've been set to the proper values. If the user provides `ResourceId` or `InputObject` with `-WhatIf`, they won't see a value for `Name` or `ResourceGroupName`